### PR TITLE
Auto configuration for spring boot

### PR DESCRIPTION
**Pull request by pavolloffay**

### CODE REVIEW

**Comment by pavolloffay** [comment]
cc @mabn @objectiser @jpkrohling

File: opentracing-spring-web-handler-interceptor-autoconfigure/src/main/java/io/opentracing/contrib/spring/web/TracingAutoconfiguration.java [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+import io.opentracing.Tracer;
+import io.opentracing.contrib.spring.web.interceptor.SpanDecorator;
+import io.opentracing.contrib.spring.web.interceptor.TracingHandlerInterceptor;
+import io.opentracing.contrib.web.servlet.filter.TracingFilter;
+
+/**
+ * @author Pavol Loffay
+ */
+@Configuration
+@AutoConfigureOrder(Ordered.HIGHEST_PRECEDENCE)
```
Comment (jpkrohling) [inline opentracing-spring-web-handler-interceptor-autoconfigure/src/main/java/io/opentracing/contrib/spring/web/TracingAutoconfiguration.java]: This is rude :) I don't know Spring that well, but if it does what I think it does, do you really need this running before everything else? Usually, these levels are reserved for security-related components. We could argue that "monitoring" is at the same level as "security", though.

File: opentracing-spring-web-handler-interceptor-autoconfigure/src/main/java/io/opentracing/contrib/spring/web/TracingAutoconfiguration.java [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+    @ConditionalOnMissingBean(Tracer.class)
+    public Tracer noopTracer() {
+        log.severe("Tracer bean is not configured! Switching to " + NoopTracer.class.getName());
+        return NoopTracerFactory.create();
+    }
+
+    @Bean
+    public FilterRegistrationBean tracingFilter() {
+        TracingFilter tracingFilter = new TracingFilter(tracer,
+                Arrays.asList(io.opentracing.contrib.web.servlet.filter.SpanDecorator.STANDARD_TAGS));
```
Comment (jpkrohling) [inline opentracing-spring-web-handler-interceptor-autoconfigure/src/main/java/io/opentracing/contrib/spring/web/TracingAutoconfiguration.java]: Those things really need a better name... This `SpanDecorator`, for instance, could have been named `ServletFilterSpanDecorator`.

File: opentracing-spring-web-handler-interceptor-autoconfigure/src/main/java/io/opentracing/contrib/spring/web/TracingAutoconfiguration.java [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+        return filterRegistrationBean;
+    }
+
+    @Bean
+    public WebMvcConfigurerAdapter tracingHandlerInterceptor() {
+        return new WebMvcConfigurerAdapter() {
+            @Override
+            public void addInterceptors(InterceptorRegistry registry) {
+                registry.addInterceptor(new TracingHandlerInterceptor(tracer,
+                        Arrays.asList(SpanDecorator.STANDARD_TAGS)));
```
Comment (jpkrohling) [inline opentracing-spring-web-handler-interceptor-autoconfigure/src/main/java/io/opentracing/contrib/spring/web/TracingAutoconfiguration.java]: Wait, is this expected to be the same `STANDARD_TAGS` as the other `SpanDecorator`?

File: opentracing-spring-web-handler-interceptor-autoconfigure/src/main/java/io/opentracing/contrib/spring/web/TracingAutoconfiguration.java (hunk 2 above)
Comment (pavolloffay) [inline opentracing-spring-web-handler-interceptor-autoconfigure/src/main/java/io/opentracing/contrib/spring/web/TracingAutoconfiguration.java]: Agree, I will submit a following PR to rename it.

File: opentracing-spring-web-handler-interceptor-autoconfigure/src/main/java/io/opentracing/contrib/spring/web/TracingAutoconfiguration.java (hunk 3 above)
Comment (pavolloffay) [inline opentracing-spring-web-handler-interceptor-autoconfigure/src/main/java/io/opentracing/contrib/spring/web/TracingAutoconfiguration.java]: Yes, I will think of a better name in following PR. IRRC interceptor's `STANDARD_TAGS` decorator adds only logs..

File: opentracing-spring-web-handler-interceptor-autoconfigure/src/main/java/io/opentracing/contrib/spring/web/TracingAutoconfiguration.java (hunk 1 above)
Comment (pavolloffay) [inline opentracing-spring-web-handler-interceptor-autoconfigure/src/main/java/io/opentracing/contrib/spring/web/TracingAutoconfiguration.java]: It has been removed, actually order of filter and interceptor is configured at different places.

**Comment by pavolloffay** [comment]
PR updated

* removed `@AutoConfigurationOrder`
* added test for auto-configuration

**Pull request merged**